backend/src/model/entity/plantings_impl.rs

```diff
@@ -30,7 +30,22 @@ impl Planting {
             query = query.filter(layer_id.eq(id));
         }
 
+        let from = search_parameters.from;
+        let to = search_parameters.to;
+
+        query = query.filter(
+            plantings::add_date
+                .is_null()
+                .or(plantings::add_date.lt(to))
+                .and(
+                    plantings::remove_date
+                        .is_null()
+                        .or(plantings::remove_date.gt(from)),
+                ),
+        );
```

Comment:
This is pretty difficult to read. Maybe you could introduce variables here to make it clearer? (Untested and bad variable names, but you get the point). It not wrong like it currently is so if my suggestion doesn't work don't spend more than a few min on it.
```suggestion
        let lower_date = plantings::add_date
                .is_null()
                .or(plantings::add_date.lt(to));
        let other_condition = plantings::remove_date
                        .is_null()
                        .or(plantings::remove_date.gt(from));
        query = query.filter(lower_date.and(other_condition));
```

Reply:
This worked great, thanks :)